src/modules/ekf2/EKF/control.cpp

```diff
@@ -1147,6 +1154,27 @@ void Ekf::controlHeightFusion()
 			}
 		}
 
+		// If no EV is available, switch to RF if possible
+		if (!_control_status.flags.ev_hgt && !_control_status.flags.rng_hgt && _range_sensor.isDataHealthy()) {
+			setControlRangeHeight();
+			fuse_height = true;
+
+			if (_control_status_prev.flags.rng_hgt != _control_status.flags.rng_hgt) {
```

Comment:
Using RF as the preferred backup probably works best when EV is being used because we would be closer to the ground, however uneven terrain under the vehicle, eg stairs will cause problems if moving. Should the commander make the vehicle stop moving if this happens?

Reply:
I would be best if the user could select the order that we fall back. Sometimes falling back to Baro will be better.

Reply:
I agree, that falling back to RF could cause issues if the terrain is not even. However, relying on baro e.g. indoors - in my experience - is worse. Main target is to give the EV subsystem some  time to re-initialize(0.8-1.5secs in my case). In combination with https://github.com/PX4/PX4-Autopilot/pull/17980 tests went pretty well falling back to RF and optical flow when EV re-initializes.

Reply:
We can't accept an upstream change that only works indoors over flat floors. Please enable the fallback option to be selectable.

Reply:
A fallback parameter then should not only control the recovery from baro (this PR) but also the fallback, which is already merged with https://github.com/PX4/PX4-ECL/pull/994